### PR TITLE
Add GRU

### LINKED ISSUE #14698: GRU

### PR DESCRIPTION
**Pull request by karalleyna**
Closes #14698

### CODE REVIEW

File: ivy/functional/ivy/experimental/layers.py [hunk 1]
```diff
@@ -1842,3 +1842,100 @@ def adaptive_avg_pool2d(
...
+    ret
+        A tuple consisting of an array that concats all the intermediate output
+        values of the hidden and the last output value of the hidden.
+    """
+    seq_length, batch_size = x.shape[0:2]
+    hidden_size = r.shape[-1]
+
+    h_t = ivy.zeros((batch_size, hidden_size)) if initial_h is None else initial_h
+
+    y = ivy.empty([seq_length, batch_size, hidden_size])
```
Comment (sherry30) [inline ivy/functional/ivy/experimental/layers.py:1910]: can remove this statement

File: ivy/functional/ivy/experimental/layers.py [hunk 2]
```diff
@@ -1842,3 +1842,100 @@ def adaptive_avg_pool2d(
...
+    - ht = g(Xt*(Wh^T) + (rt (.) Ht-1)*Rh + Rbh + Wbh)
+    - ht = g(Xt*(Wh^T) + (rt (.) (Ht-1*Rh + Rbh) + Wbh)
+    - Ht = (1 - zt) (.) ht + zt (.) Ht-1
+
+    where f is sigmoid and g is tanh.
+
+    Parameters
+    ----------
+
+    self
```
Comment (sherry30) [inline ivy/functional/ivy/experimental/layers.py:1875]: change to `x`,
same with the rest of the parameters

**Review by sherry30 — CHANGES_REQUESTED** [review]
Hi,

Your PR looks good, the tests are also passing.
I think we should also add `linear_before_reset` as a boolean argument and `sequence_lens` argument.
Let me know what you think and if you have any questions.

Thanks

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**